nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/hadoop/SecurityUtil.java

```diff
@@ -141,4 +143,37 @@ public static boolean isSecurityEnabled(final Configuration config) {
         Validate.notNull(config);
         return KERBEROS.equalsIgnoreCase(config.get(HADOOP_SECURITY_AUTHENTICATION));
     }
+
+    public static <T> T callWithUgi(UserGroupInformation ugi, PrivilegedExceptionAction<T> action) throws IOException {
```

Comment:
It doesn't seem like we really need a utility method here just to call one line `return ugi.doAs(action);` , I think it would be simpler to just call `ugi.doAs` from all the places where this method is being called

Reply:
It handles the `InterruptedException`. Most of the callers only declare `IOException`. This is a general pattern which could use a utility method in my opinion. Just in this case the wrapping would need to be replicated in 15 places or so.

Reply:
Ok fair enough